File: Java/src/main/java/com/adaptionsoft/games/uglytrivia/Array6.java
package com.adaptionsoft.games.uglytrivia;

public class Array6<T> {

    private Object[] array = new Object[6];

    public Array6(T init) {
        for (int i = 0; i < 6; i++) {
            set(i, init);
        }
    }

    public void set(int i, T value) {
        array[i] = value;
    }

    @SuppressWarnings("unchecked")
    public T get(int i) {
        return (T) array[i];
    }

}

File: Csharp/Trivia/Array6.cs
namespace UglyTrivia
{
    public class Array6<T>
    {
        private object[] array = new object[6];

        public Array6(T init)
        {
            for (int i = 0; i < 6; i++)
            {
                this.Set(i, init);
            }
        }

        public void Set(int i, T value)
        {
            this.array[i] = value;
        }

        public T Get(int i)
        {
            return (T)this.array[i];
        }
    }
}
